src/Microsoft.AspNetCore.Authorization/AuthorizeFailure.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Encapsulates a failure result of <see cref="IAuthorizationService.AuthorizeAsync(ClaimsPrincipal, object, IEnumerable{IAuthorizationRequirement})"/>.
+    /// </summary>
+    public class AuthorizeFailure
```

Comment:
I started with that, but we went with AuthenticateResult (there that was meant to signify that it was a result for the Authenticate method).  I could go either way here since I've had it both ways in this code already :)